src/Timecode.php

```diff
@@ -63,11 +67,37 @@ public function __construct(int $hours, int $minutes, int $seconds, int $frames,
      */
     public static function createFromString(string $timecodeStr, float $framerate = self::DEFAULT_FRAMERATE) : self
     {
-        list($hours, $minutes, $seconds, $frames) = explode(':', $timecodeStr);
-
-        // @codingStandardsIgnoreStart
-        return new self((int) $hours, (int) $minutes, (int) $seconds, (int) $frames, $framerate);
-        // @codingStandardsIgnoreEnd
+        // Support "hours:minutes:seconds:frames" notation
+        $matches = [];
+        preg_match('@^(?P<hours>^\d{1,3}):(?P<minutes>\d{1,2}):(?P<seconds>\d{1,2}):(?P<frames>\d{1,2})$@', $timecodeStr, $matches);
+        if (!empty($matches['hours']) && !empty($matches['minutes']) && !empty($matches['seconds']) && !empty($matches['frames'])) {
+            // @codingStandardsIgnoreStart
+            return new self(
+                $matches['hours'],
+                $matches['minutes'],
+                $matches['seconds'],
+                $matches['frames'],
+                $framerate
+            );
+            // @codingStandardsIgnoreEnd
+        }
+
+        // Support "hours:minutes:seconds.milliseconds" notation
+        $matches = [];
+        preg_match('@^(?P<hours>^\d{1,3}):(?P<minutes>\d{1,2}):(?P<seconds>\d{1,2})\.(?P<ms>\d{1,3})$@', $timecodeStr, $matches);
```

Comment:
I will prefer you split this method in 2. 
One for frame and another one for milliseconds. 
It will be more simple to test and your méthode will be determinist